src/components/huskelapp/HuskelappModal.tsx

```diff
@@ -33,88 +52,146 @@ const ModalContent = styled(Modal.Body)`
   }
 `;
 
-const StyledSkeletonWrapper = styled(SkeletonShadowbox)`
-  margin: 1em;
-  height: 5em;
-`;
-
 const HuskelappSkeleton = () => {
   return (
-    <StyledSkeletonWrapper>
+    <SkeletonShadowbox className={"m-4 h-20"}>
       <Skeleton variant="text" width="80%" />
       <Skeleton variant="text" width="30%" />
-    </StyledSkeletonWrapper>
+    </SkeletonShadowbox>
   );
 };
 
 export const HuskelappModal = ({ isOpen, toggleOpen }: HuskelappModalProps) => {
   const { huskelapp, isLoading, isSuccess } = useGetHuskelappQuery();
-  const [tekst, setTekst] = useState<string>(huskelapp?.tekst ?? "");
+  const [oppfolgingsgrunn, setOppfolgingsgrunn] = useState<Oppfolgingsgrunn>();
+  const [isFormError, setIsFormError] = useState<boolean>(false);
   const oppdaterHuskelapp = useOppdaterHuskelapp();
   const removeHuskelapp = useRemoveHuskelapp();
 
-  const oppdaterTekst = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
-    setTekst(e.target.value);
+  const oppfolgingsgrunnToText = () => {
+    switch (huskelapp?.oppfolgingsgrunn) {
+      case Oppfolgingsgrunn.AVVENT_DIALOGMOTE:
+        return texts.oppfolgingsgrunn.avventDialogmote;
+      case Oppfolgingsgrunn.VURDER_DIALOGMOTE_SENERE:
+        return texts.oppfolgingsgrunn.vurderDialogmoteSenere;
+      case Oppfolgingsgrunn.FOLG_OPP_FRA_LEGE:
+        return texts.oppfolgingsgrunn.folgOppFraLege;
+      case Oppfolgingsgrunn.FOLG_OPP_FRA_ARBEIDSGIVER:
+        return texts.oppfolgingsgrunn.folgOppFraArbeidsgiver;
+      case Oppfolgingsgrunn.TA_KONTAKT:
+        return texts.oppfolgingsgrunn.taKontakt;
+      case Oppfolgingsgrunn.VURDER_TILTAK_BEHOV:
+        return texts.oppfolgingsgrunn.vurderTiltakBehov;
+      case Oppfolgingsgrunn.ANNEN_OPPFOLGNING:
+        return texts.oppfolgingsgrunn.annenOppfolgning;
+    }
   };
 
-  const handleOppdaterHuskelapp = () => {
-    const huskelappDto: HuskelappRequestDTO = { tekst: tekst };
-    oppdaterHuskelapp.mutate(huskelappDto, {
-      onSuccess: () => toggleOpen(false),
-    });
+  const handleOnSubmit = (e: any) => {
+    e.preventDefault();
+    if (oppfolgingsgrunn !== undefined) {
+      const huskelappDto: HuskelappRequestDTO = {
+        oppfolgingsgrunn: oppfolgingsgrunn,
+      };
+      oppdaterHuskelapp.mutate(huskelappDto, {
+        onSuccess: () => toggleOpen(false),
+      });
+    } else {
+      setIsFormError(true);
+      return undefined;
+    }
+  };
+
+  const handleOppfolgingsgrunnChange = (oppfolgingsgrunn: Oppfolgingsgrunn) => {
+    setOppfolgingsgrunn(oppfolgingsgrunn);
+    setIsFormError(false);
   };
 
   const handleRemoveHuskelapp = (uuid: string) => {
     removeHuskelapp.mutate(uuid, {
       onSuccess: () => toggleOpen(false),
     });
   };
+  const isSavedHuskelapp = !!huskelapp;
```

Comment:
Synes det var litt uklart navn. Betyr dette at det finnes en eksisterende huskelapp eller at man har fått lagret en huskelapp?

Reply:
`huskelapp` er resultatet av get huskelapp requesten, så tanken var at hvis den ikke er undefined så har man en eksisterende (lagret) huskelapp, hvis ikke så får man opp valg slik at man kan lage ny 😊
Men lurer på om denne er unødvendig. Lenger nede er vi nødt til å bruke `!!huskelapp` direkte for å få inferred at den har en verdi.
Har du andre forslag/tanker? 😊

Reply:
Kan endre `const hasHuskelapp = !!huskelapp;` el. og så bare bruke den videre nedover

Reply:
Ser ut som `const hasHuskelapp = !!huskelapp` gir inferred verdi på `huskelapp` der den brukes likevel 🙏